### PR TITLE
use freds logger

### PR DESCRIPTION
**Pull request by RdrOfTheSt0rm**
Send messages going to System.out to fred's logger instead

### CODE REVIEW

File: src/plugins/Library/Library.java [hunk 1]
```diff
@@ -293,7 +293,7 @@ public synchronized void saveState(){
 						if (!(ce instanceof ExpectedMIMEEvent)) { return; }
 						synchronized(c) {
 							String type = ((ExpectedMIMEEvent)ce).expectedMIMEType;
-							System.out.println("Expected type in index: "+type);
+							Logger.error(this, "Expected type in index: "+type);
```
Comment (Bombe) [inline src/plugins/Library/Library.java]: This doesn’t really seem to be an error condition, does it have to be logged as error?

File: src/plugins/Library/SpiderIndexUploader.java [hunk 2]
```diff
@@ -288,7 +287,7 @@ private void maybeMergeToFreenet(MutableBoolean maxDiskEntrySizeExceeded) {
         synchronized(freenetMergeSync) {
             while(freenetMergeRunning) {
                 if(pushBroken) return;
-                System.err.println("Need to merge to Freenet, but last merge not finished yet. Waiting...");
+                Logger.error(this, "Need to merge to Freenet, but last merge not finished yet. Waiting...");
```
Comment (Bombe) [inline src/plugins/Library/SpiderIndexUploader.java]: Again, error seems to be too harsh here.

File: src/plugins/Library/SpiderIndexUploader.java [hunk 3]
```diff
@@ -549,29 +545,28 @@ protected void mergeToFreenet(File diskDir) {
...
 		if(f == null) {
             if(diskDir.list().length == 0) {
-                System.err.println("Directory "+diskDir+" is empty. Nothing to merge.");
+                Logger.debug(this, "Directory "+diskDir+" is empty. Nothing to merge.");
                 diskDir.delete();
                 return;
             }
             // Ignore
-            System.err.println("Unable to merge old data "+diskDir);
+            Logger.error(this, "Unable to merge old data "+diskDir);
```
Comment (Bombe) [inline src/plugins/Library/SpiderIndexUploader.java]: If this is really an error condition, shouldn’t the error be logged at the place where the error was first encountered? This does not look like the right place.

File: src/plugins/Library/SpiderIndexUploader.java [hunk 4]
```diff
@@ -848,14 +839,14 @@ public void run() {
...
-			System.out.println("Found "+dirsToMerge.length+" disk trees of old index data to merge...");
+			Logger.debug(this, "Found "+dirsToMerge.length+" disk trees of old index data to merge...");
 			Runnable r = new Runnable() {
 
 				public void run() {
 					synchronized(freenetMergeSync) {
 						while(freenetMergeRunning) {
 							if(pushBroken) return;
-							System.err.println("Need to merge to Freenet, but last merge not finished yet. Waiting...");
+							Logger.error(this, "Need to merge to Freenet, but last merge not finished yet. Waiting...");
```
Comment (Bombe) [inline src/plugins/Library/SpiderIndexUploader.java]: Again, this is more an info/normal than an error.

File: src/plugins/Library/SpiderIndexUploader.java (hunk 3 above)
Comment (RdrOfTheSt0rm) [inline src/plugins/Library/SpiderIndexUploader.java]: I will look into this i think your right again

Comment (Thynix) [inline src/plugins/Library/SpiderIndexUploader.java]: Fixing that seems independent of avoiding stdout / stderr.

**Comment by Thynix** [comment]
Thanks!

**Pull request merged**